pkg/lib/infra/sms/custom/client.go

```diff
@@ -223,30 +223,37 @@ func (c *CustomClient) Send(ctx context.Context, opts smsapi.SendOptions) error
 func handleResponse(gatewayType string, responseBody *ResponseBody, dumpedResponse []byte) error {
 
 	err := &smsapi.SendError{
+		ProviderType:   config.SMSProviderCustom,
 		DumpedResponse: dumpedResponse,
 	}
-	if responseBody.ProviderName != "" {
-		err.ProviderName = responseBody.ProviderName
-	} else {
-		err.ProviderName = gatewayType
-	}
-	if responseBody.ProviderErrorCode != "" {
-		err.ProviderErrorCode = responseBody.ProviderErrorCode
+
+	// Handle info returned by authgear-sms-gateway
+	if responseBody.Info != nil {
+		if providerName, ok := responseBody.Info["provider_name"].(string); ok {
```

Comment:
It does not handle `"provider_type"`. It should store that to CustomProviderType, and include it as attribute in metrics.

The following is an evidence.
<img width="918" height="377" alt="SCR-20251031-kzbb" src="https://github.com/user-attachments/assets/8182011f-7f96-498d-834e-13c4d71e8f1c" />